### PR TITLE
ruby: fix installation

### PR DESCRIPTION
**Pull request by iMichka**
Fixes:
```
==> Pouring ruby-2.7.1.x86_64_linux.bottle.tar.gz
Error: The `brew link` step did not complete successfully
The formula built, but is not symlinked into /home/linuxbrew/.linuxbrew
Could not symlink bin/bundle
Target /home/linuxbrew/.linuxbrew/bin/bundle
is a symlink belonging to ruby. You can unlink it:
  brew unlink ruby

To force the link and overwrite all conflicting files:
  brew link --overwrite ruby

To list all files that would be deleted:
  brew link --overwrite --dry-run ruby

Possible conflicting files are:
/home/linuxbrew/.linuxbrew/bin/bundle -> /home/linuxbrew/.linuxbrew/Cellar/ruby/2.7.1/libexec/gembin/bundle
/home/linuxbrew/.linuxbrew/bin/bundler -> /home/linuxbrew/.linuxbrew/Cellar/ruby/2.7.1/libexec/gembin/bundler
```

- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/linuxbrew-core/blob/master/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/linuxbrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
- [ ] Have you included the output of `brew gist-logs <formula>` of the build failure if your PR fixes a build failure. Please quote the exact error message.

-----

### CODE REVIEW

**Comment by iMichka** [comment]
Please have a look at 30e23470104. I think the strategy used there was not the best one.
`keg_only :provided_by_macos` is now a no-op on Linux, no need for this special path anymore.

I am unsure if I should remove the remaining bin/bundle folder though? People might have installed stuff in there?

File: Formula/ruby.rb [hunk 1]
```diff
@@ -114,6 +110,14 @@ def install
...
 
   def post_install
+    unless OS.mac?
+      # Since 2018 and commit 30e23470104,
+      # we have installed stuff in HOMEBREW_PREFIX/"bin/bundle"
+      # This is not needed anymore. Cleanup up after us.
+      rm_f HOMEBREW_PREFIX/"bin/bundle"
+      rm_f HOMEBREW_PREFIX/"bin/bundler"
+    end
+
```
Comment (sjackman) [inline Formula/ruby.rb:120]: I'm inclined to remove this chunk. The user can remove these files manually if they wish to.

**Comment by sjackman** [comment]
cc @xu-cheng

File: Formula/ruby.rb (hunk 1 above)
Comment (iMichka) [reply Formula/ruby.rb:120]: Ok

Comment (sjackman) [reply Formula/ruby.rb:120]: This chunk is still there.

File: Formula/ruby.rb [hunk 2]
```diff
@@ -38,11 +38,7 @@ def api_version
   end
 
   def rubygems_bindir
-    if OS.mac?
-      HOMEBREW_PREFIX/"lib/ruby/gems/#{api_version}/bin"
-    else
-      HOMEBREW_PREFIX/"bin"
-    end
+    HOMEBREW_PREFIX/"lib/ruby/gems/#{api_version}/bin"
```
Comment (xu-cheng) [inline Formula/ruby.rb:41]: Why this change? Noted that the `HOMEBREW_PREFIX/"lib/ruby/gems/#{api_version}/bin"` is used in Homebrew to prevent gem binaries exposed for a keg only formula. Since this formula is not keg only for Linuxbrew, we should keep gem binary path at `HOMEBREW_PREFIX/"bin"`.

Comment (iMichka) [reply Formula/ruby.rb:41]: The issue is that when you uninstall the formula, the folders stays there. And `brew link ruby` later complains about this, see above. Should we not just install to plain "#{bin}" then ?

Comment (xu-cheng) [reply Formula/ruby.rb:41]: Ok.

> Should we not just install to plain "#{bin}" then ?

We should not. The gems are support to survive across ruby formula upgrades.

Comment (xu-cheng) [reply Formula/ruby.rb:41]: This underlying issue is that users are required to manually update the versioned path `HOMEBREW_PREFIX/"lib/ruby/gems/#{api_version}/bin"` in their PATH environment whenever there is a major ruby update. That's why I choose to install them to `HOMEBREW_PREFIX/"bin"` back then.

I send a PR to Homebrew (https://github.com/Homebrew/homebrew-core/pull/53201), which should address the above issue while allowing this formula behavioring in consistent with formula in Homebrew.

**Comment by iMichka** [comment]
Just to be clear about the keg_only change `keg_only :provided_by_macos` is a no-op on Linux, so this does not change anything, ruby is not keg_only on Linux.

File: Formula/ruby.rb (hunk 2 above)
Comment (iMichka) [reply Formula/ruby.rb:41]: I am not sure I understood everything about the discussion in homebrew-core. Looks like having gems in an unversioned path is not expected. So for each ruby update you loose your gems, and you need to update your PATH accordingly?

Comment (xu-cheng) [reply Formula/ruby.rb:41]: Yes, which I think is quite cumbersome. Especially considering there may be multiple PATH in user’s configure needed to be updated. That’s why I submit the above PR though it’s rejected.

Comment (iMichka) [reply Formula/ruby.rb:41]: Let's use upstream's solution until we have a better way.
